docs/data-sources/iotda_devices.md

```diff
@@ -0,0 +1,122 @@
+
+-> When accessing an IoTDA **standard** or **enterprise** edition instance, you need to specify the IoTDA service
+  endpoint in `provider` block.
+  You can login to the IoTDA console, choose the instance **Overview** and click **Access Details**
+  to view the HTTPS application access address. An example of the access address might be
+  **9bc34xxxxx.st1.iotda-app.ap-southeast-1.myhuaweicloud.com**, then you need to configure the
+  `provider` block as follows:
+
+  ```hcl
+  provider "huaweicloud" {
+    endpoints = {
+      iotda = "https://9bc34xxxxx.st1.iotda-app.ap-southeast-1.myhuaweicloud.com"
+    }
+  }
+  ```
+
+## Example Usage
+
+```hcl
+variable "device_id" {}
+
+data "huaweicloud_iotda_devices" "test" {
+  device_id = var.device_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the devices.
+  If omitted, the provider-level region will be used.
+
+* `space_id` - (Optional, String) Specifies the space ID of the devices to be queried.
+  If omitted, query the devices in all spaces under the current instance.
+
+* `product_id` - (Optional, String) Specifies the ID of the product to be queried.
+  If omitted, query the devices in all products under the current instance.
+
+* `gateway_id` - (Optional, String) Specifies the gateway ID of the devices to be queried;
+  The `gateway_id` is the ID of the parent device to which the devices belong.
+
+* `is_cascade` - (Optional, Bool) Specifies whether to cascade queries, this parameter only takes effect when
+  carrying `gateway_id` simultaneously. The default value is **false**.  
+  The valid values are as follows:
+  + **true**: Represents querying all levels of sub devices under a device with a device ID equal to gateway ID.
+  + **false**: Represents the first level child devices under the device with the query device ID equal to gateway ID.
+  
+* `node_id` - (Optional, String) Specifies the node ID of the device to be queried.
+
+* `node_type` - (Optional, String) Specifies the node type of the devices to be queried.  
+  The valid values are as follows:
+  + **ENDPOINT**: Non-directly connected devices.
+  + **GATEWAY**: Directly connected devices or gateways.
+  + **UNKNOWN**: Unknown.
+
+* `status` - (Optional, String) Specifies the status of the devices to be queried.  
+  The valid values are as follows:
+  + **ONLINE**: Device is online.
+  + **OFFLINE**: Device is offline.
+  + **ABNORMAL**: Device is abnormal.
+  + **INACTIVE**: Device is inactive.
+  + **FROZEN**: Device is frozen.
+
+* `device_id` - (Optional, String) Specifies the ID of the device to be queried.
+
+* `name` - (Optional, String) Specifies the name of the device to be queried.
+
+* `start_time` - (Optional, String) Specifies the start time to be queried. The query result shows devices created after
+  this time (Including devices created at this time).
```

Comment:
```suggestion
  this time (including devices created at this time).
```

Reply:
It is better to give an example for fields `start_time` and `end_time`.

Reply:
fixed.